### ISSUE TITLE
How to get image to render correctly

### ISSUE THREAD

**Issue by jeremygottfried** [martinlaxenaire/curtainsjs#31]
Hi, I'm using your node package with react. 
I am trying to set up the basic plane example, but for some reason, my image is not rendering on the canvas. 
It is only rendering a black canvas that makes waves in response to my mouse movements.
How can I get my image to render?

```js
import React, { useRef, useEffect } from 'react';
import curtain from '../../assets/images/curtain.jpg'
import './curtain.scss';
import { Curtains } from 'curtainsjs';
import vertexShader from './vert.js';
import fragmentShader from './frag.js';
import { XYDelta, linearInterpolate } from '../../utils/physics';

function damp(a, b, dampFactor) {
  return (a - b) * dampFactor
}

export default function Curtain() {
  const mouseDamp = 10;
  const warpDepth = 1;
  const decayCoefficient = 0.3;
  const rippleWidth = 0.02;
  const curtainElement = useRef(null);
  const curtainContainer = useRef(null);
  const plane = useRef(null);
  const lerpedPosition = useRef({
    x: 0,
    y: 0,
  })
  const prevLerpedPosition = useRef({
    x: 0,
    y: 0,
  })
  const deltas = useRef({
    max: 0,
    applied: 0,
  })

  useEffect(() => {
    // set up our WebGL context and append the canvas to our wrapper
    const webGLCurtain = new Curtains({
        container: "canvas",
        watchScroll: false
    });

    webGLCurtain
    .onError(
      function(e) {
        console.error(e);
      }
    )
    .onContextLost(
      function() {
        webGLCurtain.restoreContext();
      }
    )

    const { clientWidth, clientHeight } = curtainElement.current;
    const { x: mouseX, y: mouseY } = lerpedPosition.current;
    const params = {
        vertexShader,
        fragmentShader,
        widthSegments: 20,
        heightSegments: 20,
        uniforms: {
            resolution: { // resolution of our plane
                name: "uResolution",
                type: "2f",
                value: [clientWidth, clientHeight],
            },
            time: { // time uniform that will be updated at each draw call
                name: "uTime",
                type: "1f",
                value: 0,
            },
            mousePosition: { // our mouse position
                name: "uMousePosition",
                type: "2f", // again an array of floats
                value: [mouseX, mouseY],
            },
            mouseMoveStrength: { // the mouse move strength
                name: "uMouseMoveStrength",
                type: "1f",
                value: 0,
            }
        }
    };
    const _plane = webGLCurtain.addPlane(curtainElement.current, params);
    _plane
    .onReady(
      function() {
          // set a fov of 35 to exagerate perspective
          _plane.setPerspective(35);

          // apply a little effect once everything is ready
          deltas.current.max = 2;
          plane.current = _plane;
      }
    )
    .onRender(
      function() {
          const { uniforms: { time, mouseMoveStrength } } = _plane;
          // increment our time uniform
          time.value++;

          // decrease both deltas by damping : if the user doesn't move the mouse, effect will fade away
          deltas.current.applied -= damp(deltas.current.applied, deltas.current.max, rippleWidth);
          deltas.current.max -= damp(deltas.current.max, 0, rippleWidth);

          // send the new mouse move strength value
          mouseMoveStrength.value = deltas.current.applied;
      }
    )
    .onAfterResize(
      function() {
          const { uniforms: { resolution } } = _plane;
          const { width, height } = _plane.getBoundingRect();
          resolution.value = [width, height];
      }
    )

    return () => {
      plane.current = null;
      webGLCurtain.dispose();
    }
  })

  function handleMovement(clientX, clientY) {
    if (plane.current) {
      const { x: prevX, y: prevY } = prevLerpedPosition.current = lerpedPosition.current
      const { x: lerpX, y: lerpY } = lerpedPosition.current = {
        x: linearInterpolate(prevX, clientX, decayCoefficient),
        y: linearInterpolate(prevY, clientY, decayCoefficient)
      }
      const { x, y } = plane.current.mouseToPlaneCoords(lerpX, lerpY);
      plane.current.uniforms.mousePosition.value = [x,y];

      if (prevX && prevY) {
        const delta = XYDelta(lerpedPosition.current, prevLerpedPosition.current, mouseDamp, warpDepth)
        if (delta >= deltas.current.max) {
          deltas.current.max = delta;
        }
      }
    }
  }

  function onMouseMove({ clientX, clientY }) {
    handleMovement(clientX, clientY);
  }

  function onTouchMove({ targetTouches: [{ clientX, clientY }] }) {
    handleMovement(clientX, clientY);
  }

  // function onMouseMove() {}
  // function onTouchMove() {}
  return <div
           ref={curtainContainer}
           id="page-wrap"
           onMouseMove={onMouseMove}
           onTouchMove={onTouchMove}
          >
      <div id="canvas"/>
      <div ref={curtainElement} className="curtain">
        <img src={curtain} alt='banner' />
      </div>
    </div>
}
```
```css
#page-wrap {
    width: 100%;
    height: 100%;
    position: relative;
}

/*** canvas ***/

#canvas {
    height: 100%;
    width: 100%;
}


.curtain {
    position: absolute;
    top: 0;
    width: 100%;
    height: 100%;
}

.curtain img {
    width: 100%;
    height: 100%;
    display: none;
}
```

**Comment by martinlaxenaire** [comment]
Hi @jeremygottfried,

if your plane is rendering a black texture, it means there's been an error while loading your image.
What's the javascript console saying?
Also double check how you're importing your image, there might be an error there:

```javascript
import curtain from '../../assets/images/curtain.jpg'
```

Cheers,

**Comment by jeremygottfried** [comment]
Hi @martinlaxenaire ,
Thank you for the quick reply!
I checked the console and there is no error being logged, which is why I came here. 
I can also see that the image element is loaded in the plane's images array.
I'm using the same image you used here: https://www.curtainsjs.com/examples/simple-plane/index.html
I really appreciate the help.

**Comment by jeremygottfried** [comment]
Ah I figured it out. My image was missing the `data-sampler` attribute. It would be helpful to add some documentation saying "data-sampler" is the attribute used by the shader to identify your image. I just wasted 3 hours debugging this.

**Comment by martinlaxenaire** [comment]
Fixed in v5.3:
- Updated the Texture class documentation in order to make the use of the data-sampler attribute clearer

**Comment by jeremygottfried** [comment]
Thank you!
